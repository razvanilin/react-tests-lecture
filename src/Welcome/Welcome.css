.welcomeText{
  margin-top: 100px;
}
.welcomeButton{
  margin-top: 50px;
}
