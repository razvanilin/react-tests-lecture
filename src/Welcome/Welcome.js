import React, { Component } from "react";
import { Link } from "react-router-dom";
import "./Welcome.css";

class Welcome extends Component {
  render() {
    return (
      <div className="Welcome">
        <div className="row">
          <div className="col"></div>
          <div className="col-5 text-center">
            <div className="welcomeText">
            Lorem ipsum dolor sit amet, an mei impetus dissentiunt. Reque mundi no eos, sea lorem eripuit pericula ad, mel aperiam voluptatum in. Mei suas verear instructior ea. Eu has incorrupte instructior.
            </div>
            <div className="welcomeButton">
              <Link to="/search">
                <button type="button" className="btn btn-warning">Create new dinner</button>
              </Link>
            </div>
          </div>
          <div className="col"></div>
        </div>
      </div>
    );
  }
}

export default Welcome;
