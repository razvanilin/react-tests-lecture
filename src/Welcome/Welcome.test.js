import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";

import Welcome from "./Welcome";

it("The Welcome component renders correctly", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Welcome />
      </MemoryRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
