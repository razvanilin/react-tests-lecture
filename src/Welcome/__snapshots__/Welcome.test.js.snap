// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Welcome component renders correctly 1`] = `
<div
  className="Welcome"
>
  <div
    className="row"
  >
    <div
      className="col"
    />
    <div
      className="col-5 text-center"
    >
      <div
        className="welcomeText"
      >
        Lorem ipsum dolor sit amet, an mei impetus dissentiunt. Reque mundi no eos, sea lorem eripuit pericula ad, mel aperiam voluptatum in. Mei suas verear instructior ea. Eu has incorrupte instructior.
      </div>
      <div
        className="welcomeButton"
      >
        <a
          href="/search"
          onClick={[Function]}
        >
          <button
            className="btn btn-warning"
            type="button"
          >
            Create new dinner
          </button>
        </a>
      </div>
    </div>
    <div
      className="col"
    />
  </div>
</div>
`;
